nucypher-core-wasm/tests/wasm.rs

```diff
@@ -663,3 +663,87 @@ fn metadata_response() {
         "MetadataResponse does not roundtrip"
     );
 }
+
+//
+// ThresholdDecryptionRequestResponse
+//
+
+#[wasm_bindgen_test]
+fn threshold_decryption_request() {
+    let ritual_id: u16 = 5;
+    let request_secret = SecretKey::random();
+    let request_encrypting_key = request_secret.public_key();
+
+    let response_secret = SecretKey::random();
+    let response_encrypting_key = response_secret.public_key();
+
+    let conditions: JsValue = Some(Conditions::new("{'some': 'condition'}")).into();
+    let context: JsValue = Some(Context::new("{'user': 'context'}")).into();
+
+    let request = ThresholdDecryptionRequest::new(
+        ritual_id,
+        0,
+        b"fake ciphertext",
+        &conditions.unchecked_into::<OptionConditions>(),
+        &context.unchecked_into::<OptionContext>(),
+    )
+    .unwrap();
+
+    let encrypted_request = request.encrypt(&request_encrypting_key, &response_encrypting_key);
+
+    let encrypted_request_bytes = encrypted_request.to_bytes();
+    let encrypted_request_from_bytes =
+        EncryptedThresholdDecryptionRequest::from_bytes(&encrypted_request_bytes).unwrap();
+
+    assert_eq!(encrypted_request_from_bytes.id(), ritual_id);
```

Comment:
Are we testing to bytes here? Any reason not to check against the `encrypted_request`?
```suggestion
    assert_eq!(encrypted_request_from_bytes.id(), encrypted_request);
```

Reply:
✔️ I'm only mimicking the encrypted request going over the wire by serializing, then deserializing it.

I can add the bytes assertion, but I still want to keep the ritual id check.